### PR TITLE
claude: Further develop rules/api.md; add testing.md and design-variables.md; add `DesignVariables.folderText`

### PR DESCRIPTION
**Pull request by chrisbobbe**
It's particularly exciting to think we can automate the task of adding to `DesignVariables`!

### CODE REVIEW

File: .claude/CLAUDE.md [hunk 1]
```diff
@@ -67,6 +67,7 @@ Uses Pigeon for type-safe platform channels (Android intents, notifications).
 - **No `dart format`** — follow existing code style manually. Auto-format is disabled in VS Code settings.
 - **Prefer relative imports** within the package.
 - **Strict analysis**: strict-inference, strict-raw-types, strict-casts are all enabled.
+- **Stream → channel rename**: Use "channel" instead of "stream" for new variable/parameter names (e.g., `channelId` not `streamId`). The API still uses "stream" in many places, but new code in this codebase should use "channel". See #631.
```
Comment (gnprice) [inline .claude/CLAUDE.md:70]: Confusingly, Claude put this in
    ~/.claude/projects/-Users-chrisbobbe-dev-zulip-flutter/memory/MEMORY.md
    instead of in the repo.
    
    I entered the following prompt:
    
    > I'm confused. Shouldn't that have gone in this zulip-flutter
    > project folder, in .claude? I want to commit the result.
    
    to which Claude replied:
    
    ⏺ That file _is_ in the project folder — it's at
      .claude/projects/-Users-chrisbobbe-dev-zulip-flutter/memory/MEMORY.md
      under my home directory, which is my per-project auto memory. That
      persists across my sessions but it's not in the repo, so you can't
      commit it.
    
      If you want this in the repo so the whole team benefits, I should
      add it to the existing .claude/CLAUDE.md in the repo. Let me do
      that.

Here's the Claude docs for the different types of memory:
https://code.claude.com/docs/en/memory#determine-memory-type

So that doc says "auto memory" for the place the tool initially put this info, and "project memory" for what you intended.

Possibly the thing to do here is, in the initial prompt, to ask specifically to put it in "project memory".

File: .claude/rules/design-variables.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
+# Adding a Color to DesignVariables
```
Comment (gnprice) [inline .claude/rules/design-variables.md]: (As we discussed in the office, I think this can be skipped in favor of letting Claude study that one source file directly when it comes up.)

File: .claude/rules/testing.md [hunk 3]
```diff
@@ -0,0 +1,304 @@
+# Testing Conventions
```
Comment (gnprice) [inline .claude/rules/testing.md:6]: This rule can probably benefit from a bit of frontmatter with paths. See api.md.

File: lib/widgets/theme.dart [hunk 4]
```diff
@@ -450,6 +453,7 @@ class DesignVariables extends ThemeExtension<DesignVariables> {
     return extension!;
   }
 
+  // Named variables from the Figma, perhaps with different capitalization.
   final Color background;
   final Color bannerBgIntDanger;
```
Comment (gnprice) [inline lib/widgets/theme.dart]: Huh. Is the capitalization irregular? It's camel case here, and IIRC it's kebab case in the Figma.

If we can say the exact pattern explicitly (and succinctly), that's probably best.

**Comment by chrisbobbe** [comment]
Thanks for the review! Revision pushed.

File: .claude/rules/testing.md [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+
+Descriptions are **short fragments** (not sentences), lowercase, no trailing period:
+```dart
+group('handleMessageEvent', () {
+  test('smoke', () { ... });
+  test('with starred flag', () { ... });
+});
+```
+
+Group nesting is typically 2–3 levels deep. The top level groups by feature or method name; deeper levels group by subcategory or edge case.
```
Comment (gnprice) [inline .claude/rules/testing.md]: It looks like "2-3 levels deep" is accurate if you count the test case as one of the levels. I did a quick survey:
```
$ git grep -oh '^\s+group' test/ | sort -r | uniq -c
    304   group
    168     group
     58       group
     14         group

$ git grep -oh '^\s+test' test/ | sort -r | uniq -c
    312   test
      1    test
   1093     test
    713       test
    201         test
     48           test
     23             test
      2               test
      3                 test
```

So about 75% of test cases are indented either 1 or 2 levels deeper than the top-level body of the `main` function.

The wording seems ambiguous, though, and I'm not sure Claude will always interpret it the same way it did when it wrote it. (The way I first read it is the opposite: 2-3 levels of groups, then the test case. Which I think would feel excessive.) So let's be explicit.

**Comment by chrisbobbe** [comment]
Thanks for the review! Revision pushed, with that tweak.

> There was a task you'd tried earlier (this morning?) where you had some notes with a list of things Claude didn't get right. How does it do if you retry it with these changes?

Ah I'm afraid I've lost those notes; probably best if I don't spend time rerunning this particular experiment. For the feature I was working on, it's probably best if I/we take over the existing PR #1956 rather than asking Claude to start from `main`.

File: .claude/settings.json [hunk 6]
```diff
@@ -7,6 +7,7 @@
       "Bash(find:*)",
       "WebFetch(domain:zulip.com)",
       "Bash(tools/icons/build-icon-font:*)",
+      "Bash(dart run build_runner build:*)",
```
Comment (chrisbobbe) [inline .claude/settings.json]: Hmm is this what we want? See 2e552b4f

Comment (gnprice) [reply .claude/settings.json]: Hmm true — maybe drop this commit, then.

**Comment by gnprice** [comment]
Thanks! Sounds good re: the previous experiment.

**Comment by chrisbobbe** [comment]
Thanks for the review! Revision pushed, dropping a commit as suggested.
